src/osd/PGBackend.cc

```diff
@@ -447,7 +447,7 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
 		  << (known ? "known" : "best guess")
 		  << " omap_digest 0x" << auth.omap_digest << std::dec
 		  << " from auth shard " << auth_shard;
-      result.set_omap_digest_mismatch();
+      result.set_omap_digest_mismatch(known);
```

Comment:
@dzafman 
Is the only difference between the case where we have no oi digest but the shards do not match each other and the case where we do have the oi.digest and at least one shard does not match that digest that in the latter case at least one shard will have have a digest error?  I don't think that's good -- I'd rather the top level error list distinguish between those cases explicitly so that the user of this interface knows what to look for.